task-sdk/tests/task_sdk/execution_time/test_supervisor.py

```diff
@@ -1490,6 +1491,7 @@ def watched_subprocess(self, mocker):
     def test_handle_requests(
         self,
         mock_secrets_masker,
+        mock_mask_secret,
```

Comment:
Why do we need both mock_secrets_masker and mock_mask_secret? That feels like we over mocking -- I don't think we should need both. Could we assert something on `mock_secrets_masker.return_value` without having to add an extra mock? Or can we get rid of one?

Reply:
Yeah let me try and simplify it